packages/fuels-core/src/types/transaction_builders.rs

```diff
@@ -392,48 +450,76 @@ impl CreateTransactionBuilder {
     }
 }
 
-fn convert_to_fuel_inputs(inputs: &[Input], offset: usize) -> Vec<FuelInput> {
-    let mut new_offset = offset;
-
+/// Resolve SDK Inputs to fuel_tx Inputs. This function will calculate the right
+/// data offsets for predicates and set witness indexes for signed coins.
+fn resolve_fuel_inputs(
```

Comment:
The fn is starting to be a bit long (82 lines). 

Wdyt about splitting it up a bit? Maybe flattening the nesting level by removing the inner match on `resource`?